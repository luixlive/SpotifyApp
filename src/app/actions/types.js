// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
