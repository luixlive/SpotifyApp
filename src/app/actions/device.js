import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
