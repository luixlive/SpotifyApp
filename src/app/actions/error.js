import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
