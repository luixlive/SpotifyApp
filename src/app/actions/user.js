import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
