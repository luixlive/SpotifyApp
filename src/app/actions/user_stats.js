import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
