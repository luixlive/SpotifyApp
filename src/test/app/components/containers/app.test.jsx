import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import {
  app as componentProps,
} from './../../../test_utils/components_props';
import {
  ConnectedApp,
  PureApp,
} from './../../../../app/components/containers/app';
import initialState from './../../../test_utils/initial_state';
import injectRouter from './../../../test_utils/inject_router';
import trivialReactElement from './../../../test_utils/trivial_react_element';
import * as types from './../../../../app/actions/types';

describe('App Components - App', () => {
  const mockStore = configureStore();
  let store;
  beforeAll(() => {
    store = mockStore(initialState);
  });

  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders user not loaded', () => {
      const rendered = renderer.create(injectRouter(() => (
        <Provider store={store}>
          <PureApp {...props} />
        </Provider>
      ))).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders user loaded', () => {
      props.userLoaded = true;
      const rendered = renderer.create(injectRouter(() => (
        <Provider store={store}>
          <PureApp {...props} />
        </Provider>
      ))).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = mount(injectRouter(() => (
        <Provider store={store}>
          <ConnectedApp>
            {trivialReactElement}
          </ConnectedApp>
        </Provider>
      )));
    });

    it('renders', () => {
      expect(wrapper.find(PureApp).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureApp).props()))
        .toEqual(Object.keys(componentProps));
    });

    it('dispatches loadUser', () => {
      const expectedAction = {
        type: types.LOAD_USER,
        payload: { },
      };
      expect(store.getActions()[0]).toEqual(expectedAction);
    });
  });
});

