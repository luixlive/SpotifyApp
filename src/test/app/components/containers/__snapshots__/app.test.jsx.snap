// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - App Snapshots renders error 1`] = `
<div>
  <div
    className="ui green center aligned segment"
  >
    <div
      className="ui three column grid"
    >
      <div
        className="four wide column"
      />
      <div
        className="middle aligned eight wide column"
      >
        <h1
          className="ui header"
        >
          Spotify Stats
        </h1>
      </div>
      <div
        className="four wide column"
      >
        <button
          className="ui green inverted right floated button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Log in with Spotify
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui center aligned container"
  >
    <h1
      className="ui header"
      style={
        Object {
          "fontSize": "4em",
          "marginTop": "1.5em",
        }
      }
    >
      Error :(
    </h1>
    <h2
      className="ui header"
    >
      
       Go back to
      <a
        href="/"
        onClick={[Function]}
      >
         Home
      </a>
      .
    </h2>
  </div>
  <div
    className="ui center aligned segment footer"
  >
    <p>
      Favicon made by
      <a
        href="https://www.flaticon.com/authors/gregor-cresnar"
        rel="noopener noreferrer"
        target="_blank"
      >
         Gregor Cresnar 
      </a>
      from
      <a
        href="https://www.flaticon.com"
        rel="noopener noreferrer"
        target="_blank"
      >
         www.flaticon.com
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`App Components - App Snapshots renders user loaded 1`] = `
<div>
  <div
    className="ui green center aligned segment"
  >
    <div
      className="ui three column grid"
    >
      <div
        className="four wide column"
      />
      <div
        className="middle aligned eight wide column"
      >
        <h1
          className="ui header"
        >
          Spotify Stats
        </h1>
      </div>
      <div
        className="four wide column"
      >
        <button
          className="ui green inverted right floated button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Log in with Spotify
        </button>
      </div>
    </div>
  </div>
  <div />
  <div
    className="ui center aligned segment footer"
  >
    <p>
      Favicon made by
      <a
        href="https://www.flaticon.com/authors/gregor-cresnar"
        rel="noopener noreferrer"
        target="_blank"
      >
         Gregor Cresnar 
      </a>
      from
      <a
        href="https://www.flaticon.com"
        rel="noopener noreferrer"
        target="_blank"
      >
         www.flaticon.com
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`App Components - App Snapshots renders user not loaded 1`] = `
<div>
  <div
    className="ui green center aligned segment"
  >
    <div
      className="ui three column grid"
    >
      <div
        className="four wide column"
      />
      <div
        className="middle aligned eight wide column"
      >
        <h1
          className="ui header"
        >
          Spotify Stats
        </h1>
      </div>
      <div
        className="four wide column"
      >
        <button
          className="ui green inverted right floated button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Log in with Spotify
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui basic segment screen-loader"
  >
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="center"
        >
          <div
            className="ui big inverted text loader"
          >
            <h3
              className="ui header"
            >
              Loading...
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui center aligned segment footer"
  >
    <p>
      Favicon made by
      <a
        href="https://www.flaticon.com/authors/gregor-cresnar"
        rel="noopener noreferrer"
        target="_blank"
      >
         Gregor Cresnar 
      </a>
      from
      <a
        href="https://www.flaticon.com"
        rel="noopener noreferrer"
        target="_blank"
      >
         www.flaticon.com
      </a>
      .
    </p>
  </div>
</div>
`;
